### PR TITLE
[core] Migrate many_nodes_actor_tests to new cloud.

### PR DESCRIPTION
**Pull request by fishbone**
## Why are these changes needed?

This PR make the test run with the new cloud to prevent regression.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/ray_release/command_runner/command_runner.py [hunk 1]
```diff
@@ -85,7 +87,12 @@ def run_prepare_command(
         Command runners may choose to run this differently than the
         test command.
         """
-        return self.run_command(command, env, timeout)
+        return exponential_backoff_retry(
```
Comment (rkooo567) [inline release/ray_release/command_runner/command_runner.py:90]: what is this change for?

**Review by rkooo567 — APPROVED** [review]
what do you mean by the regression btw?

File: release/ray_release/command_runner/command_runner.py (hunk 1 above)
Comment (iycheng) [reply release/ray_release/command_runner/command_runner.py:90]: Somehow in the new stack it needs waiting for a while (1-3s) before it's ready to be used

**Comment by iycheng** [comment]
> what do you mean by the regression btw? 

The infra can't start 2k nodes and hangs and oss ray somehow GCS disappear. :(

**Comment by rkooo567** [comment]
can you run the release test before merging it?

**Comment by iycheng** [comment]
@rkooo567 I triggered the test hours ago (https://buildkite.com/ray-project/release-tests-pr/builds/26327#0185e53f-d293-4e2a-a34b-b6aa149e30ab) but still pending :(
I'll come back later.

**Comment by iycheng** [comment]
passed

**Pull request merged**